### PR TITLE
PWN-6878 - update cashout transaction after moonpay

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6878

## Description of Work

update cashout transaction after moonpay

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt [hunk 1]
```diff
@@ -56,6 +56,16 @@ class SellPayloadFragment :
         }
     }
 
+    override fun onStart() {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt:59]: heh =(

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt (hunk 1 above)
Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt:59]: ?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadFragment.kt:59]: can we call presenter method to update data only on onResume?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt [hunk 2]
```diff
@@ -42,5 +42,7 @@ interface SellPayloadContract {
         fun cashOut()
         fun onTokenAmountChanged(newValue: String)
         fun onUserMaxClicked()
+        fun onStart()
+        fun onStop()
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt:46]: Can you name this methods with more context? 
For example, `checkSellLock` or smth

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt [hunk 3]
```diff
@@ -78,6 +79,7 @@ class SellPayloadPresenter(
     }
 
     private suspend fun checkForSellLock() {
+        needCheckForSellLock = false
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:82]: What does it mean? 
Sell Lock? Does it mean the user is blocked for selling?
Maybe you'll call it `isSellAvailable` then?

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt [hunk 4]
```diff
@@ -95,6 +97,14 @@ class SellPayloadPresenter(
         }
     }
 
+    override fun onStart() {
+        launch { if (needCheckForSellLock) checkForSellLock() }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:101]: You could move `if (needCheckForSellLock)` inside `checkForSellLock` function

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt [hunk 5]
```diff
@@ -95,6 +97,14 @@ class SellPayloadPresenter(
...
+    }
+
+    override fun onStop() {
+        needCheckForSellLock = true
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:105]: I cannot get, why are you changing this value on stop. 
What is the idea?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:82]: we have a fragment `SellLocked` when the user is locked from the sell 
so we check that is Sell is locked for user, naming is ok here, i approve

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt (hunk 2 above)
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt:46]: we can implement DefaultLifecycleObserver presenter contract, to have this methods as Default

**Review by Davran331122 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt (hunk 2 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadContract.kt:46]: we don't need it, it's rare case

File: app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt (hunk 5 above)
Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:105]: `checkForSellLock()`call on `attach` `org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:65` and have comment `// call order is important!`
also we need to call `checkForSellLock()` only after return from moonpay webView for `navigateToSellLock` `org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt:89`

so if fragment livecycle only onStop()-> onStart() need to call `checkForSellLock`
if fragment recreated i don't want to call `checkForSellLock` twice `needCheckForSellLock` is control it

**Review by Davran331122 — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**